### PR TITLE
Key storage updates

### LINKED ISSUE #153: [cPP CHANGE] FCS_STG_EXT.2 & 3
FCS_STG_EXT.2 refers to FCS_COP.1/KeyEnc, which is no longer a thing.

I've been trying to figure out a way to get rid of FCS_STG_EXT.2 & 3 and replace them with FCS_CKM_EXT.3. I haven't quite gotten there, but I suggest the following change:

FCS_STG_EXT.2.1. The TSF shall encrypt [AKs, SKs, KEKs, and [selection: long-term trusted channel key material, all software-based key storage, no other keys]] using one of the following methods: [selection: key encapsulation, key wrapping].

There are a couple problems:
At this point key wrapping does not include GCM and CCM modes, but maybe the CWG will include them. This would also eliminate the need for FCS_STG_EXT.3, since all those methods provide integrity as well as confidentiality.

The other problem is that the old FCS_COP.1/KeyEnc allows "encryption" through XOR of a key with a KEK. This isn't allowed by the new catalog. Nor does it provide integrity. It would make things a lot simpler if we could get rid of  XOR because then all the allowed methods would include integrity and we could get rid of STG_EXT.3.

### PR DESCRIPTION
**Pull request by woodbe**
This is to close #153.

### CODE REVIEW

File: cPP_DSC.adoc [hunk 1]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Cryptographic key usage in cryptographic operations* - Depending on the application, a cryptographic key could be root key which by virtue of being a root key, cannot be encrypted with another key.  It is often stored in plain sight, sometimes with hardware protections, sometimes obfuscated in clever ways, such as in physically unclonable functions.  Other than root keys, it is expected that other cryptographic keys should be protected with encryption, often using other keys, which leads to a chain or hierarchy of cryptographic keys that encrypt other keys often anchored by a root key.   Additionally, it is expected that an entity authenticate itself self as an authorized user of the key.  This latter requirement is in addition to the recommended requirements of the Common Criteria Part 2 for FCS_CKM.3._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: editorial: ... itself self ...

Comment (smuellerDD) [inline cPP_DSC.adoc]: Regarding the last sentence: What exactly should that statement imply - any special actions for the ST author?

File: cPP_DSC.adoc [hunk 2]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Cryptographic key recovery* - This refers to the retrieval of cryptographic keys from either archival, backup, or escrow locations.  In each case, the TOE uses the agreed upon cryptographic key access method agreed upon._
++
+*_Modifications to attributes of cryptographic keys* - Often during the creation of cryptographic keys, the TOE assigns certain attributes to them.  This may include allowed key access methods, authentication methods, and other items as discussed above in Creation of cryptographic keys.  The TOE may require authentication for authorization to modify these attributes._
++
+*_Cryptographic key destruction* - The attributes of the cryptographic key should indicate the necessary conditions needed to destroy it.  Often this will be an authentication indicating authorization to destroy it.  This SFR would apply to keys stored in the TOE, and FCS_CKM.6 would apply for destruction.  Otherwise, the destruction of keys stored outside the TOE is not enforceable by this SFR._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: editorial: ... FCS_CKM.6 would apply ...
-> shall apply ?

File: cPP_DSC.adoc [hunk 3]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++ 
+*_Cryptographic key archival, backup, and escrow* - TOEs often perform cryptographic key archival to manage limited memory resources inside its own security boundaries.  Often the TOE encrypts the cryptographic keys prior to saving them to storage that is close by on the same device as the TOE and that can be accessed quickly. TOEs may perform cryptographic key backups into storage that is meant for longer term keeping.  Backup storage is often not on the same device as the TOE and may be physically hundreds of miles away.  In practice, the TOE encrypts the keys using an approved method before sending them to backup storage.  TOEs may perform cryptographic key escrow in which it entrusts a third party with access to the private or secret keys.  In practice, the TOE protects the keys using a cryptographic key access method agreed upon with the escrow agent before sending them._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: On escrow: Should the escrow mechanism be linked to some objective for the environment or OSP considering that the TOE has no means to enforce this?

File: cPP_DSC.adoc [hunk 4]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Cryptographic key destruction* - The attributes of the cryptographic key should indicate the necessary conditions needed to destroy it.  Often this will be an authentication indicating authorization to destroy it.  This SFR would apply to keys stored in the TOE, and FCS_CKM.6 would apply for destruction.  Otherwise, the destruction of keys stored outside the TOE is not enforceable by this SFR._
++
+*_Key Access Methods_*
++
+_This SFR lists several methods for protecting the confidentiality of keys prior to authorizing their use, archival, backup, escrow, or recovery.  This protection is often afforded only to the private or secret portion of cryptographic keys.  Some methods offer integrity protection as well.  These methods may not apply to modification of attributes especially if the attributes do not need confidentiality.  These methods may not apply to destruction since a TOE can only destroy keys within its boundaries, and it is premused these keys are already in the boundary._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: On key destruction outside the TOE boundary: Should we link that to some objective of the environment or OSP?

File: cPP_DSC.adoc [hunk 5]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Key encapsulation* - If the STF claims key encapsulation, then it should use approved methods such as those listed in CKM.2._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: editorial: ... STF ... -> SFR?

Comment (smuellerDD) [inline cPP_DSC.adoc]: editorial: ... should use ... -> must use?

File: cPP_DSC.adoc [hunk 6]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Key wrapping* - If the STF claims key wrapping, then it should use approved methods such as KW, or KWP._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: STF -> SFR

Comment (smuellerDD) [inline cPP_DSC.adoc]: again: should -> must?

File: cPP_DSC.adoc [hunk 7]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Key encryption* - If the STF claims key encryption, then it should use approved methods such as those in FCS_COP.1/SK._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: STF-> SFR

Comment (smuellerDD) [inline cPP_DSC.adoc]: again: should -> must?

File: cPP_DSC.adoc [hunk 8]
```diff
@@ -1023,6 +1023,94 @@ FCS_CKM.2.1:: The TSF shall distribute cryptographic keys in accordance with a s
...
++
+*_Authentication for Key Access Methods_*
++
+_The PP/ST should contain FIA SFRs for authenticating users.  See NIST SP 800-63B “Digital Identity Guidelines” for guidance on methods for authenticating users for access to keys for any purpose._
```
Comment (smuellerDD) [inline cPP_DSC.adoc]: should -> must?

**Comment by woodbe** [comment]
@smuellerDD all the text above comes straight from the crypto catalog. I do think we should discuss the specific list of "what you can do" and the accompanying descriptions. I actually think we may want to restrict the options available from the list in the catalog, but I also suspect that the catalog text may change some (as you note, several of these seem to be language issues or a lack of clarity).

@jdonndelinger do you have an update to the catalog we could look at that may resolve some of these?

**Comment by woodbe** [comment]
Merged based on 10/12 call

**Pull request merged**